better_profanity/better_profanity.py

```diff
@@ -116,11 +121,25 @@ def _count_non_allowed_characters(self, word):
         return count
 
     def _generate_patterns_from_word(self, word):
-        """Return all patterns can be generated from the word."""
+        """
+        Return all patterns can be generated from the word. Returns an empty tuple if
+        the word has too many variants.
+        """
         combos = [
             (char,) if char not in self.CHARS_MAPPING else self.CHARS_MAPPING[char]
             for char in iter(word)
         ]
+
+        # Prevent exponential memory consumption runoff.
+        num_patterns = reduce(operator.mul, [len(chars) for chars in combos], 1)
+        if num_patterns > MAX_PATTERNS:
+            print(
```

Comment:
Instead of `print`, what do you think about using a logger ?
Right now, the lib does not have a logger, so you could create one on top of `better_profanity.py`:
```
logger = logging.getLogger('better_profanity')
logger.setLevel(logging.WARNING)
handler = logging.StreamHandler(stderr)
handler.setFormatter(
                logging.Formatter(
                    fmt="%(levelname)s [%(asctime)s] %(name)s - %(message)s",
                    datefmt="%Y-%m-%d %H:%M:%S",
                )
            )
logger.addHandler(handler)
```

and call `logger.warning`

Reply:
I like this approach best, yeah.